src/component_container.jl

```diff
@@ -32,25 +31,22 @@ function get_components end
 "Get the single component that matches the given specification from the `ComponentContainer`, or `nothing` if there is no match."
 function get_component end
 
-"Get whether the given component of the given system is available for use (defaults to true)."
-get_available(::ComponentContainer, ::InfrastructureSystemsComponent) = true
-
 "Like `get_components` but only on components that are available."
 function get_available_components end
 
 "Like `get_component` but only on components that are available."
 function get_available_component end
 
 get_available_components(sys::ComponentContainer, args...; kwargs...) =
-    get_components(c -> get_available(sys, c), sys, args...; kwargs...)
+    get_components(c -> get_available(c), sys, args...; kwargs...)
```

Comment:
get_available always returns true/false. There is no need to create a new anonymous function 
```suggestion
    get_components(get_available, sys, args...; kwargs...)
```